autode/conformers/conformers.py

```diff
@@ -234,6 +237,65 @@ def prune_diff_graph(
         logger.info(f"Pruned on connectivity {n_prev_confs} -> {len(self)}")
         return None
 
+    def prune_with_boltzmann(
+        self, temperature: float = 298.15, threshold: float = 0.9
+    ) -> None:
+        """
+        Keep the lowest energy conformers whose cumulative weights are up to
+        the threshold (default 90%). Conformer energies must be in Ha.
+        -----------------------------------------------------------------------
+
+        Arguments:
+            threshold (float): Cumlative weight threshold.
+            temperature (float): Temperature in K.
+        """
+        assert threshold <= 1.0, "The treshold must be below 1.0"
+
+        hartrees_to_joules = 4.35974417 * 10**-18
+        normalisation_constant = 0.0
+        minimum_energy = self.lowest_energy.energy
+
+        for conf in self:
+            assert conf.energy.units == ha, "Conformer energy must be in Ha"
+
+            normalised_energy = conf.energy - minimum_energy
+            energy_joules = normalised_energy * hartrees_to_joules
+            conf.boltzmann_weight = exp(
+                -energy_joules / (temperature * Boltzmann)
+            )
+            normalisation_constant += conf.boltzmann_weight
+
+        # Sort conformers in descending order by Boltzmann weight
+        self.sort(key=lambda conf: conf.boltzmann_weight, reverse=True)
```

Comment:
The function sorting the set of conformers is not obvious – appreciate it makes the rest of the function more difficult 😅

Reply:
Not sure I follow - would you mind clarifying?

Reply:
yeah sorry – rubbish explanation. 
for example, if I've got a list `l = [2, 1, 5, None]` and I call a function called `prune_not_none` on it I'd expect 
```python
>>> prune_not_none(l)
[2, 1, 5]
```
and not be `[1, 2, 5]`